pyodide_build/pypabuild.py

```diff
@@ -79,7 +89,15 @@ def _runner(cmd, cwd=None, extra_environ=None):
 
         # Some build dependencies like cmake, meson installs binaries to this directory
         # and we should add it to the PATH so that they can be found.
-        env["BUILD_ENV_SCRIPTS_DIR"] = isolated_build_env.scripts_dir
+        if isolated_build_env:
+            env["BUILD_ENV_SCRIPTS_DIR"] = isolated_build_env.scripts_dir
+        else:
+            # For non-isolated builds, set a fallback path or use the current Python path
+            import sysconfig
+
+            scripts_dir = sysconfig.get_path("scripts")
+            env["BUILD_ENV_SCRIPTS_DIR"] = scripts_dir
```

Comment:
I am curious if this is adapted from pypa/build.

Reply:
Yes, sort of. `pypa/build` has a more robust check for where the `bin/` dir is usually placed. It's here: https://github.com/pypa/build/blob/811c9db0dcac14b7b32f08ee57d09af3666979d4/src/build/env.py#L336-L376. I used a subset of it through `sysconfig.get_path("scripts")`, as we don't need to support all sorts of Python distributions that might be packaged differently.

Reply:
Makes sense. Thanks for the explanation!